tools/integration_tests/read_cache/job_chunk_test.go

```diff
@@ -0,0 +1,221 @@
+		expectedOutcome[0], expectedOutcome[1] = expectedOutcome[1], expectedOutcome[0]
+		testFileNames[0], testFileNames[1] = testFileNames[1], testFileNames[0]
+	}
+	assert.Equal(t, expectedOutcome[0].BucketName, structuredJobLogs[0].BucketName)
+	assert.Equal(t, expectedOutcome[1].BucketName, structuredJobLogs[1].BucketName)
+	assert.Equal(t, expectedOutcome[0].ObjectName, structuredJobLogs[0].ObjectName)
+	assert.Equal(t, expectedOutcome[1].ObjectName, structuredJobLogs[1].ObjectName)
+	if s.isLimitedByMaxParallelDownloads {
+		require.LessOrEqual(t, chunkCount, int64(len(structuredJobLogs[0].JobEntries)))
+		for i := 0; int64(i) < int64(len(structuredJobLogs[0].JobEntries)); i++ {
+			offset := min(s.chunkSize*int64(i+1), fileSize)
+			assert.GreaterOrEqual(t, offset, structuredJobLogs[0].JobEntries[i].Offset)
+		}
+	} else {
+		require.EqualValues(t, chunkCount, len(structuredJobLogs[0].JobEntries))
+		for i := 0; int64(i) < chunkCount; i++ {
+			offset := min(s.chunkSize*int64(i+1), fileSize)
+			assert.Equal(t, offset, structuredJobLogs[0].JobEntries[i].Offset)
+		}
+	}
+}
+
+////////////////////////////////////////////////////////////////////////
+// Test Function (Runs once before all tests)
+////////////////////////////////////////////////////////////////////////
+
+func TestJobChunkTest(t *testing.T) {
+	ts := &jobChunkTest{ctx: context.Background()}
+	// Create storage client before running tests.
+	closeStorageClient := client.CreateStorageClientWithTimeOut(&ts.ctx, &ts.storageClient, 15*time.Minute)
+	defer func() {
+		err := closeStorageClient()
+		if err != nil {
+			t.Errorf("closeStorageClient failed: %v", err)
+		}
+	}()
+
+	// Run tests for mounted directory if the flag is set.
+	if setup.AreBothMountedDirectoryAndTestBucketFlagsSet() {
+		test_setup.RunTests(t, ts)
+		return
+	}
+
+	var cacheSizeMB int64 = 48
+
+	// Tests to validate chunk size when read cache parallel downloads are disabled.
+	var chunkSizeForReadCache int64 = 8
+	ts.flags = []string{"--config-file=" + createConfigFile(cacheSizeMB, true, configFileName, false)}
+	ts.chunkSize = chunkSizeForReadCache * util.MiB
+	log.Printf("Running tests with flags: %s", ts.flags)
+	test_setup.RunTests(t, ts)
+
+	// Tests to validate chunk size when read cache parallel downloads are enabled
+	// with unlimited max parallel downloads.
+	ts.flags = []string{"--config-file=" +
+		createConfigFileForJobChunkTest(cacheSizeMB, false, "unlimitedMaxParallelDownloads", parallelDownloadsPerFile, maxParallelDownloads, downloadChunkSizeMB)}
+	ts.chunkSize = parallelDownloadsPerFile * downloadChunkSizeMB * util.MiB
+	log.Printf("Running tests with flags: %s", ts.flags)
+	test_setup.RunTests(t, ts)
+
+	// Tests to validate chunk size when read cache parallel downloads are enabled
+	// with go-routines not limited by max parallel downloads.
+	parallelDownloadsPerFile := 4
+	maxParallelDownloads := 24 // maxParallelDownloads = parallelDownloadsPerFile * downloadChunkSizeMB * number of files
```

Comment:
nit: maxParallelDownloads := 24 // maxParallelDownloads = parallelDownloadsPerFile * downloadChunkSizeMB * number of files

should be just
maxParallelDownloads := 9 // maxParallelDownloads > parallelDownloadsPerFile * number of files being accessed simultanesouly.